plugins/azure-functions/src/components/OverviewTableComponent/OverviewTable.tsx

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import { Box, Card, Link, LinearProgress } from '@material-ui/core';
+import { FunctionsData } from '../../api/types';
+import { Table, TableColumn } from '@backstage/core-components';
+import FlashOnIcon from '@material-ui/icons/FlashOn';
+
+type States = 'Waiting' | 'Running' | 'Paused' | 'Failed';
+
+const State = ({ value }: { value: States }) => {
+  const colorMap = {
+    Waiting: '#dcbc21',
+    Running: 'green',
+    Paused: 'black',
+    Failed: 'red',
+  };
+  return (
+    <Box display="flex" alignItems="center">
+      <span
+        style={{
+          display: 'block',
+          width: '8px',
+          height: '8px',
+          borderRadius: '50%',
+          backgroundColor: colorMap[value],
+          marginRight: '5px',
+        }}
+      />
+      {value}
+    </Box>
+  );
+};
+
+type FunctionTableProps = {
+  data: FunctionsData[];
+  loading: boolean;
+};
+
+const DEFAULT_COLUMNS: TableColumn<FunctionsData>[] = [
+  {
+    title: 'name',
+    highlight: true,
+    render: (func: FunctionsData) => {
+      return (
+        <Link href={func.href} target="_blank">
+          {func.functionName}
+        </Link>
+      );
+    },
+  },
+  {
+    title: 'location',
+    render: (func: FunctionsData) => func.location ?? 'unknown',
+  },
+  {
+    title: 'status',
+    render: (func: FunctionsData) => <State value={func.state as States} />,
```

Comment:
There's actually a set of components in `@backstage/core-components` that handles this for you, here's the Storybook examples: https://backstage.io/storybook/?path=/story/data-display-status--default

Reply:
Strange enough, this doesn't work.

I get this error:

> TypeError: Cannot read property 'ok' of undefined

      58 |     },
      59 |     pending: {
    > 60 |       '&::before': {
         |                    ^
      61 |         backgroundColor: theme.palette.status.pending,
      62 |       },
      63 |     },

            at Object.<anonymous>.useStyles.name (../../../packages/core-components/src/components/Status/Status.tsx:60:20)
      at Object.create (../../../node_modules/@material-ui/styles/getStylesCreator/getStylesCreator.js:32:35)
      at attach (../../../node_modules/@material-ui/styles/makeStyles/makeStyles.js:113:32)
      at ../../../node_modules/@material-ui/styles/makeStyles/makeStyles.js:258:7
      at useSynchronousEffect (../../../node_modules/@material-ui/styles/makeStyles/makeStyles.js:211:14)
      at useStyles (../../../node_modules/@material-ui/styles/makeStyles/makeStyles.js:250:5)
      at StatusOK (../../../packages/core-components/src/components/Status/Status.tsx:97:16)
      at renderWithHooks (../../../node_modules/react-dom/cjs/react-dom.development.js:14985:18)
      at mountIndeterminateComponent (../../../node_modules/react-dom/cjs/react-dom.development.js:17811:13)
      at beginWork (../../../node_modules/react-dom/cjs/react-dom.development.js:19049:16)
      at HTMLUnknownElement.callCallback (../../../node_modules/react-dom/cjs/react-dom.development.js:3945:14)
      at HTMLUnknownElement.callTheUserObjectsOperation (../../../node_modules/jsdom/lib/jsdom/living/generated/EventListener.js:26:30)
      at innerInvokeEventListeners (../../../node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:338:25)
      at invokeEventListeners (../../../node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:274:3)
      at HTMLUnknownElementImpl._dispatch (../../../node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:221:9)
      at HTMLUnknownElementImpl.dispatchEvent (../../../node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:94:17)
      at HTMLUnknownElement.dispatchEvent (../../../node_modules/jsdom/lib/jsdom/living/generated/EventTarget.js:231:34)
      at Object.invokeGuardedCallbackDev (../../../node_modules/react-dom/cjs/react-dom.development.js:3994:16)
      at invokeGuardedCallback (../../../node_modules/react-dom/cjs/react-dom.development.js:4056:31)
      at beginWork$1 (../../../node_modules/react-dom/cjs/react-dom.development.js:23964:7)
      at performUnitOfWork (../../../node_modules/react-dom/cjs/react-dom.development.js:22779:12)
      at workLoopSync (../../../node_modules/react-dom/cjs/react-dom.development.js:22707:5)
      at renderRootSync (../../../node_modules/react-dom/cjs/react-dom.development.js:22670:7)
      at performSyncWorkOnRoot (../../../node_modules/react-dom/cjs/react-dom.development.js:22293:18)
      at scheduleUpdateOnFiber (../../../node_modules/react-dom/cjs/react-dom.development.js:21881:7)
      at updateContainer (../../../node_modules/react-dom/cjs/react-dom.development.js:25482:3)
      at ../../../node_modules/react-dom/cjs/react-dom.development.js:26021:7
      at unbatchedUpdates (../../../node_modules/react-dom/cjs/react-dom.development.js:22431:12)
      at legacyRenderSubtreeIntoContainer (../../../node_modules/react-dom/cjs/react-dom.development.js:26020:5)
      at Object.render (../../../node_modules/react-dom/cjs/react-dom.development.js:26103:10)
      at ../../../node_modules/@testing-library/react/dist/pure.js:101:25
      at batchedUpdates$1 (../../../node_modules/react-dom/cjs/react-dom.development.js:22380:12)
      at act (../../../node_modules/react-dom/cjs/react-dom-test-utils.development.js:1042:14)
      at render (../../../node_modules/@testing-library/react/dist/pure.js:97:25)


With the markup (for testing purpose):
```ts
{
    title: 'status',
    render: (func: Partial<FunctionsData>) => (
      <Box display="flex" alignItems="center">
        <StatusOK /> Running
      </Box>
    ),
},
```

Reply:
Odd, we use that exact component in the Azure DevOps plugin's BuildTable: https://github.com/backstage/backstage/blob/210cfcc0a0a2b6bb336f0ef608937bd810aab318/plugins/azure-devops/src/components/BuildTable/BuildTable.tsx#L41

And that gets called here: https://github.com/backstage/backstage/blob/210cfcc0a0a2b6bb336f0ef608937bd810aab318/plugins/azure-devops/src/components/BuildTable/BuildTable.tsx#L141

Reply:
Any luck with this?